sycl/include/CL/sycl/handler.hpp

```diff
@@ -825,9 +825,8 @@ class __SYCL_EXPORT handler {
 
       range<Dims> AdjustedRange = NumWorkItems;
       AdjustedRange.set_range_dim0(NewValX);
-#ifdef __SYCL_DEVICE_ONLY__
       kernel_parallel_for_wrapper<NameWT, TransformedArgType>(Wrapper);
```

Comment:
This is performed in both host and device, isn't it?
Why is that needed?

Reply:
@romanovvlad  made these changes, but the root of it is that we're required to have the kernel instantiated on both sides now, since otherwise we cant tell whether a lambda is used to name a kernel on the host-side.